workfinder/search/__init__.py

```diff
@@ -55,3 +62,88 @@ def download_world_borders():
         logging.info(f'Downloaded {borders_remote}')
     else:
         logging.info(f'{borders_local} already available')
+
+
+def get_redis_connection():
+
+    host = get_config("redis", "url")
+    port = get_config("redis", "port")
+
+    connection = redis.Redis(host=host, port=port, db=0)
+
+    return connection
+
+
+_s3_conn = None
+
+
+def get_s3_connection():
+    global _s3_conn
+
+    if not _s3_conn:
+
+        access = get_config("AWS", "access_key_id")
+        secret = get_config("AWS", "secret_access_key")
+        bucket_name = get_config("AWS", "bucket")
+        endpoint_url = get_config("AWS", "end_point")
+        s3_region = get_config("AWS", "region")
+
+        _s3_conn = S3Utils(access, secret, bucket_name, endpoint_url, s3_region)
+    return _s3_conn
+
+
+def list_s3_files(prefix):
+    s3 = get_s3_connection()
+    path_sizes = s3.list_files_with_sizes(prefix)
+    return path_sizes
+
+
+def list_catalog(collection_path):
+    s3 = get_s3_connection()
+    catalog_body = s3.get_object_body(collection_path)
+    catalog = json.loads(catalog_body.decode('utf-8'))
+    collection = Collection.from_dict(catalog)
+    result = [i.id for i in collection.get_items()]
+    return result
+
+
+_nc = NATS()
```

Comment:
not a really fan of global variables, but you can anyway get a much cleaner code creating this session objects from the entrypoint part of your code